syllabus/curricula/plr/task_sampler.py

```diff
@@ -261,12 +272,93 @@ def _sample_unseen_level(self):
 
         return task
 
+    def compute_returns(self, gamma, gae_lambda, rewards, value_preds, masks):
+        assert self.requires_value_buffers, "Selected strategy does not use compute_rewards."
+        gae = 0
+        returns = torch.zeros_like(rewards)
+        for step in reversed(range(rewards.size(0))):
+            delta = (
+                    rewards[step]
+                    + gamma * value_preds[step + 1] * masks[step + 1]
+                    - value_preds[step]
+            )
+            gae = delta + gamma * gae_lambda * masks[step + 1] * gae
+            returns[step] = gae + value_preds[step]
+        return returns
+
+    def evaluate_task(self, task, env, get_action_and_value_fn, gamma, gae_lambda):
+
+        if env is None:
+            raise ValueError("Environment object is None. Please ensure it is properly initialized.")
+        obs = env.reset(next_task=task)
+        done = False
+        episode_data = {
+            'tasks': [],
+            'masks': [],
+            'rewards': [],
+            'returns': [],
+            'value_preds': [],
+            'policy_logits': []
+        }
+
+        while not done:
+            action, value = get_action_and_value_fn(obs)
+            obs, rew, done, info = env.step(action)
+
+            episode_data['tasks'].append(task)
+            episode_data['masks'].append(not done)
+            episode_data['rewards'].append(rew)
+            episode_data['value_preds'].append(value)
+            episode_data['policy_logits'].append(info['policy_logits'])
+
+        episode_data['returns'] = self.compute_returns(gamma, gae_lambda, episode_data['rewards'],
+                                                       episode_data['value_preds'])
+
+        return episode_data
```

Comment:
```suggestion
    def evaluate_task(self, task, env, get_action_and_value_fn, gamma, gae_lambda):

        if env is None:
            raise ValueError("Environment object is None. Please ensure it is properly initialized.")
        obs = env.reset(next_task=task)
        done = False
        rewards = []
        masks = []
        values = []

        while not done:
            action, value = get_action_and_value_fn(obs)
            obs, rew, term, trunc, info = env.step(action)
            
            rewards.append(rew)
            masks.append(not (term or trunc))
            values.append(value)
        return = self.compute_returns(gamma, gae_lambda, rewards, values, masks)

        return {
            "tasks": task,
            "masks": masks,
            "rewards": rewards,
            "value_preds": values,
            "returns": return
        }
```
I think you can simplify the code here. Also you'll need to be careful, the environments we're using are probably Gymnasium environments not Gym environments, meaning they'll return obs, rew, term, trunc, info rather than obs, rew, done, info.